DNN Platform/Library/Entities/Portals/StyleColorBase.cs

```diff
@@ -0,0 +1,180 @@
+﻿#region Copyright
+
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE file in the project root for full license information.
+
+#endregion
+
+#region usings
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+
+#endregion
+
+
+
+namespace DotNetNuke.Entities.Portals
+{
+    /// <summary>
+    /// Represents a CSS color and it's components
+    /// </summary>
+    public class StyleColorBase
+    {
+        private enum Component
```

Comment:
I do not _think_ so, I believe it would overcomplicated stuff and the main goal here is to provide a way to set your brand guide colors. I have rarely seen brand guides that specify opacity.

However, I added r, g, b values too to the rendered css custom properties, the idea being if you need something to be in your primary brand color but transparent, you could use:

```css
.myClass {
 background-color: rgba(var(--dnn-color-primary-r), var(--dnn-color-primary-g), var(--dnn-color-primary-b), 0.5)
}
```

But open to suggestions, just wanted to keep it simple

Reply:
Okay.  👍